text/src/autogluon/text/automm/data/trivial_augmenter.py

```diff
@@ -0,0 +1,252 @@
+    if random.random() > 0.5:
+        degrees = -degrees
+    return pil_img.rotate(degrees)
+
+
+rotate = TransformT("Rotate", _rotate_impl)
+
+
+def _solarize_impl(pil_img, level):
+    """Applies PIL Solarize to `pil_img` with strength level."""
+    max = 256
+    level = scale_parameter(level, max, "int")
+    return ImageOps.solarize(pil_img, max - level)
+
+
+solarize = TransformT("Solarize", _solarize_impl)
+
+
+def _posterize_impl(pil_img, level):
+    """Applies PIL Posterize to `pil_img` with strength level."""
+    max = 4
+    min = 0
+    level = scale_parameter(level, max - min, "int")
+    return ImageOps.posterize(pil_img, max - level)
+
+
+posterize = TransformT("Posterize", _posterize_impl)
+
+
+def _enhancer_impl(enhancer):
+    """Sets level to be between 0.1 and 1.8 for ImageEnhance transforms of PIL."""
+    min = 0.1
+    max = 1.8
+
+    def impl(pil_img, level):
+        v = scale_parameter(level, max - min, "float") + min  # going to 0 just destroys it
+        return enhancer(pil_img).enhance(v)
+
+    return impl
+
+
+color = TransformT("Color", _enhancer_impl(ImageEnhance.Color))
+
+contrast = TransformT("Contrast", _enhancer_impl(ImageEnhance.Contrast))
+
+brightness = TransformT("Brightness", _enhancer_impl(ImageEnhance.Brightness))
+
+sharpness = TransformT("Sharpness", _enhancer_impl(ImageEnhance.Sharpness))
+
+
+def _shear_x_impl(pil_img, level):
+    """Shears the image along the horizontal axis with `level` magnitude."""
+    max = 0.3
+    level = scale_parameter(level, max, "float")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, level, 0, 0, 1, 0))
+
+
+shear_x = TransformT("ShearX", _shear_x_impl)
+
+
+def _shear_y_impl(pil_img, level):
+    """Shear the image along the vertical axis with `level` magnitude."""
+    max = 0.3
+    level = scale_parameter(level, max, "float")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, level, 1, 0))
+
+
+shear_y = TransformT("ShearY", _shear_y_impl)
+
+
+def _translate_x_impl(pil_img, level):
+    """Translate the image in the horizontal direction by `level` number of pixels."""
+    max = 10
+    level = scale_parameter(level, max, "int")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, level, 0, 1, 0))
+
+
+translate_x = TransformT("TranslateX", _translate_x_impl)
+
+
+def _translate_y_impl(pil_img, level):
+    """Translate the image in the vertical direction by `level` number of pixels."""
+    max = 10
+    level = scale_parameter(level, max, "int")
+    if random.random() > 0.5:
+        level = -level
+    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, 0, 1, level))
+
+
+translate_y = TransformT("TranslateY", _translate_y_impl)
+
+
+def set_image_augmentation_space(max_strength):
+    global image_parameter_max
```

Comment:
We should avoid using global variable. If we create multiple trivial augmenters, would the earlier ones overwrite the later ones?

Reply:
Yes. But they are all 30?

Reply:
Consider to use a class variable to hold this value.